public/app/plugins/datasource/loki/modifyQuery.ts

```diff
@@ -165,6 +165,18 @@ export function addLabelToQuery(
   const hasStreamSelectorMatchers = getMatcherInStreamPositions(query);
   // For non-indexed labels we want to add them after label_format to, for example, allow ad-hoc filters to use formatted labels
   const labelFormatPositions = getNodePositionsFromQuery(query, [LabelFormatExpr]);
+
+  // If the label type wasn't passed in from the calling function, we can use lezer to figure out if this label is already in the stream selectors
+  if (!labelType) {
+    const hasIdentifierSelectorMatchers = getIdentifierInStreamPositions(query);
+    const indexedKeys = hasIdentifierSelectorMatchers.map((match) => match.getExpression(query));
+    const alreadyAddedAValueForThisStreamSelectorLabel = indexedKeys.includes(key);
+
+    if (alreadyAddedAValueForThisStreamSelectorLabel) {
```

Comment:
Just a nit, but does this variable help understanding the code better? For me it's just adding more unneeded verbosity.